src/rafter_consensus_fsm.erl

```diff
@@ -841,25 +823,39 @@ become_candidate(#state{term=CurrentTerm, me=Me}=State0) ->
     _ = request_votes(State3),
     State3.
 
-become_leader(#state{me=Me, term=Term, init_config=InitConfig}=State) ->
-    NewState = State#state{leader=Me,
+become_leader(#state{me=Me, term=Term, config=Config, init_config=InitConfig, timer=Timer}=State) ->
+    NewState0 = State#state{leader=Me,
                            responses=dict:new(),
                            followers=initialize_followers(State),
                            send_clock = 0,
                            send_clock_responses = dict:new(),
-                           read_reqs = orddict:new()},
+                           read_reqs = orddict:new(),
+                           timer = Timer},
 
     case InitConfig of
         complete ->
             %% Commit a noop entry to the log so we can move the commit index
             Entry = #rafter_entry{type=noop, term=Term, cmd=noop},
-            append(Entry, NewState);
-        _ ->
-            %% First entry must always be a config entry
-            NewState
+            append(Entry, NewState0);
+        undefined ->
```

Comment:
Ahh yes, you are correct. I was thinking of the case when the leader dies after being elected but before it sends the `#append_entries{}` . In this case the config is not blank on the followers because the 1st entry is in the log already. This config gives the follower a vote [here](https://github.com/andrewjstone/rafter/blob/master/src/rafter_consensus_fsm.erl#L134) and allows the follower to become a candidate. While we should probably set `init_config=complete` when the follower first learns it has a config [here](https://github.com/andrewjstone/rafter/blob/master/src/rafter_consensus_fsm.erl#L177), this fix does seem to do the right thing when it matters, and it isn't in the hot path of `#append_entries{}` either. +1.